collectors/misc/score/src/main/java/com/capitalone/dashboard/ThresholdUtils.java

```diff
@@ -0,0 +1,312 @@
+    boolean daysPresent,
+    boolean isValueMax
+  ) {
+    String greaterOrMore = daysPresent ? "more" : "greater",
+      less = "less than ",
+      greater = greaterOrMore + " than ",
+      lessOrEqual = "less than or equal to ",
+      greaterOrEqual = greaterOrMore + " than or equal to ",
+      comparatorMessagePass = "",
+      comparatorMessageFail = "";
+
+
+    switch (comparatorType) {
+      case equals:
+        break;
+      case less:
+        comparatorMessagePass = less;
+        if (!daysPresent) {
+          comparatorMessageFail = comparatorMessagePass;
+        } else if (!isValueMax) {
+          comparatorMessageFail = greaterOrEqual;
+        }
+        break;
+      case greater:
+        comparatorMessagePass = greater;
+        if (!daysPresent) {
+          comparatorMessageFail = comparatorMessagePass;
+        } else {
+          comparatorMessageFail = lessOrEqual;
+        }
+        break;
+      case less_or_equal:
+        comparatorMessagePass = lessOrEqual;
+        if (!daysPresent) {
+          comparatorMessageFail = comparatorMessagePass;
+        } else {
+          comparatorMessageFail = greater;
+        }
+        break;
+      case greater_or_equal:
+        if (!isValueMax) {
+          comparatorMessagePass = greaterOrEqual;
+        }
+        if (!daysPresent) {
+          comparatorMessageFail = comparatorMessagePass;
+        } else {
+          comparatorMessageFail = less;
+        }
+        break;
+      default:
+        break;
+    }
+
+    return Pair.of(comparatorMessagePass, comparatorMessageFail);
+  }
+
+  public static ScoreTypeValue getScoreForThreshold(ScoreThresholdSettings percentThreshold, int compareResult) {
+    switch (percentThreshold.getComparator()) {
+      case equals:
+        if (compareResult == 0) {
+          return percentThreshold.getScore();
+        }
+        return null;
+      case less:
+        if (compareResult == -1) {
+          return percentThreshold.getScore();
+        }
+        return null;
+      case greater:
+        if (compareResult == 1) {
+          return percentThreshold.getScore();
+        }
+        return null;
+      case less_or_equal:
+        if (compareResult == 0 || compareResult == -1) {
+          return percentThreshold.getScore();
+        }
+        return null;
+      case greater_or_equal:
+        if (compareResult == 0 || compareResult == 1) {
+          return percentThreshold.getScore();
+        }
+        return null;
+      default:
+        return null;
+    }
+
+  }
+
+  public static ScoreWeight.ProcessingState getCriteriaStateFromScore(ScoreTypeValue scoreTypeValue) {
+    if (scoreTypeValue.getScoreValue().compareTo(Constants.ZERO_SCORE) > 0) {
+      return ScoreWeight.ProcessingState.criteria_passed;
+    }
+    return ScoreWeight.ProcessingState.criteria_failed;
+  }
+
+  public static String getThresholdMessage(ScoreWeight.ProcessingState state, Pair<String, String> messagePair) {
```

Comment:
@sidbha007  you can remove this method if it is not used.

Reply:
have removed it